webpack/assets/javascripts/react_app/components/HostDetails/Audits/index.js

```diff
@@ -31,93 +29,87 @@ const AuditCard = ({ hostName }) => {
       setActiveAccordion(id);
     }
   };
-  const dispatch = useDispatch();
-  useEffect(() => {
-    dispatch(
-      get({
-        key: 'HOST_DETAILS_AUDITS',
-        url: foremanUrl(`/api/audits?search=host+%3D+${hostName}`),
-      })
-    );
-  }, [hostName, dispatch]);
-  const audits = useSelector(state =>
-    selectAPIResponse(state, 'HOST_DETAILS_AUDITS')
-  );
 
+  const {
+    response: { results },
+    setUrl,
+  } = useAPI('HOST_DETAILS_AUDITS', 'get', undefined);
+
+  useEffect(() => {
+    // the hostName prop arrives by an async request within the slot component
+    if (hostName) setUrl(foremanUrl(`/api/audits?search=host+%3D+${hostName}`));
```

Comment:
could we call `useAPI` directly inside the `useEffect` instead of making the url dynamic inside the `useAPI` hook?

Reply:
hooks have to be called in the [top-level](https://reactjs.org/docs/hooks-rules.html) with no conditions or loops, that's why the `setUrl` is given to the caller.

Reply:
Thanks for the explanation!
Looking at it again, do we even need the `useEffect` then? Since the `hostName` arrives in the `props` to this component, it would trigger a re-render when it changes, so having something like:
```js
  const url = hostName && foremanUrl(`/api/audits?search=host+%3D+${hostName}`);
  const {
    response: { results },
  } = useAPI('HOST_DETAILS_AUDITS', 'get', url);
```
should work without needing the extra state inside the hook and effect in the caller to update it

Reply:
Thanks, but this doesn't seem to work, the `url` is only initialized once in the first rendering, therefore any re-rendering won't update it and it will be kept as undefined.

Reply:
Thanks, makes it simpler!  but this approach doesn't seem to work for asynchronous calls, (i.e after a setTimeout, or whether the url itself arrives from an ajax call), though I don't have any real use case for now.